langchain/src/llms/googlevertexai.ts

```diff
@@ -0,0 +1,387 @@
+import { GoogleAuth } from "google-auth-library";
+import { BaseLLM, BaseLLMParams } from "./base.js";
+import { Generation, LLMResult } from "../schema/index.js";
+
+interface GoogleVertexAiBaseLLMInput extends BaseLLMParams {
+  /** Sampling temperature to use */
+  temperature?: number;
+
+  /**
+   * Maximum number of tokens to generate in the completion.
+   */
+  maxTokens?: number;
+
+  /**
+   * Top-p changes how the model selects tokens for output.
+   *
+   * Tokens are selected from most probable to least until the sum
+   * of their probabilities equals the top-p value.
+   *
+   * For example, if tokens A, B, and C have a probability of
+   * .3, .2, and .1 and the top-p value is .5, then the model will
+   * select either A or B as the next token (using temperature).
+   */
+  topP?: number;
+
+  /**
+   * Top-k changes how the model selects tokens for output.
+   *
+   * A top-k of 1 means the selected token is the most probable among
+   * all tokens in the model’s vocabulary (also called greedy decoding),
+   * while a top-k of 3 means that the next token is selected from
+   * among the 3 most probable tokens (using temperature).
+   */
+  topK?: number;
+
+  /** Hostname for the API call */
+  endpoint?: string;
+
+  /** Region where the LLM is stored */
+  location?: string;
+
+  /** Model to use */
+  model?: string;
+}
+
+export interface GoogleVertexAiLLMInput extends GoogleVertexAiBaseLLMInput {}
+
+export interface GoogleVertexAiChatInput extends GoogleVertexAiBaseLLMInput {
+  /** Any messages from earlier in this conversation */
+  prefixMessages?: ChatMessage[];
+
+  /** Instructions how the model should respond */
+  context?: string;
+
+  /** Help the model understand what an appropriate response is */
+  examples?: ChatExample[];
+
+  /**
+   * A map of OpenAI role names and their corresponding Vertex AI
+   * author name.
+   */
+  roleAlias?: RoleAlias;
+}
+
+export const ChatMessageRoleEnum = {
+  System: "system",
+  User: "user",
+  Assistant: "assistant", // For OpenAI compatibility
+  Bot: "bot",
+} as const;
+
+export type ChatMessageRoleEnum =
+  (typeof ChatMessageRoleEnum)[keyof typeof ChatMessageRoleEnum];
+
+type RoleAlias = Record<string, string>;
+
+/**
+ * Based on OpenAI ChatCompletionRequestMessage
+ */
+export interface ChatMessage {
+  role: ChatMessageRoleEnum;
+  content: string;
+  name?: string;
+}
+
+/**
+ * Represents a single "example" exchange that can be provided to
+ * help illustrate what a model response should look like.
+ */
+export interface ChatExample {
+  input: ChatMessage;
+  output: ChatMessage;
+}
+
+interface Response {
+  data: {
+    predictions: Prediction[];
+  };
+}
+
+interface GoogleVertexAiLLMTextInstance {
+  content: string;
+}
+
+interface GoogleVertexAiChatExample {
+  input: GoogleVertexAiChatMessage;
+  output: GoogleVertexAiChatMessage;
+}
+
+interface GoogleVertexAiChatMessage {
```

Comment:
Can you try to make this match: https://github.com/hwchase17/langchainjs/blob/main/langchain/src/schema/index.ts#L65